state.c

```diff
@@ -961,6 +980,13 @@ static void pv_state_transfer_platforms(struct pv_state *pending,
 		       v->name, v->plat->name);
 		dl_list_del(&v->list);
 		dl_list_add_tail(&current->volumes, &v->list);
+
+		if (v->disk) {
+			d = pv_state_fetch_disk(current, v->disk->name);
+			pv_log(DEBUG, "relinking volume %s to disk %s", v->name,
+			       d->name);
+			v->disk = d;
```

Comment:
is this helping without making a copy of the disk from old state or am i missing something?

Reply:
the disks are already there. The current one is the one that we have used in the old revision, and the one that we are going to use the new one too. We are just linking new volumes to the old revision disk here, as the one from the new state will be removed.